interfaces/apparmor/template.go

```diff
@@ -508,3 +508,13 @@ var nfsSnippet = `
   network inet,
   network inet6,
 `
+
+// overlayRootSnippet contains the extra permissions necessary for snap and
+// snap-confine to operate on systems where '/' is a writable overlay fs.
+// AppArmor requires directory reads for upperdir (but these aren't otherwise
+// visible to the snap).
+var overlayRootSnippet = `
+  # snapd autogenerated workaround for systems using '/' on overlayfs. For
+  # details see: https://bugs.launchpad.net/apparmor/+bug/1703674
+  "###UPPERDIR###/{,**/}" r,
```

Comment:
Is  the double quoting around this part desired? I think this is a mistake.

Reply:
It is desired. If there is a space in upperdir then we need the double quote.

Reply:
Done (I added a comment on why we do it).